src/libImaging/Storage.c

```diff
@@ -258,23 +259,106 @@ ImagingDelete(Imaging im) {
 /* Allocate image as an array of line buffers. */
 
 #define IMAGING_PAGE_SIZE (4096)
+#define IMAGING_ARENA_BLOCK_SIZE (16 * 1024 * 1024)
 
+#ifdef IMAGING_TLS
+/* This is the overall process-level index that keeps track of the next index
+ * that will be assigned to a thread.
+ */
+static uint64_t ImagingArenaIndex = UINT64_MAX;
+
+/* This is the thread-local index that associated a thread with an arena in the
+ * statically-allocated list.
+ */
+static IMAGING_TLS uint64_t ImagingArenaThreadIndex = UINT64_MAX;
```

Comment:
A comment that the code below relies on `UINT64_MAX + 1` overflowing to 0 would probably be a good idea here.

Reply:
No longer necessary since we can initialize `ImagingArenaIndex` to `0` now.